typescript/packages/jumble/src/contexts/AuthenticationContext.tsx

```diff
@@ -77,44 +77,51 @@ export const AuthenticationProvider: React.FC<{ children: React.ReactNode }> = (
   //
   // Must be called within a user gesture.
   const passkeyRegister = useCallback(async (name: string, displayName: string) => {
-    return PassKey.create(name, displayName);
+    const credential = await PassKey.create(name, displayName);
+    return credential;
   }, []);
 
-
   // This should be called when a passkey (possibly) exists for the user already,
   // and no root key has yet been stored (e.g. first login). Subsequent page loads
   // should load key from storage and not require this callback.
   //
   // Must be called within a user gesture.
-  const passkeyAuthenticate = useCallback(async () => {
-    if (!keyStore) {
-      return;
-    }
-    let passkey = await PassKey.get();
-    let root = await passkey.createRootKey();
-    await keyStore.set(ROOT_KEY, root);
-    setRoot(root);
-  }, [keyStore]);
- 
+  const passkeyAuthenticate = useCallback(
+    async (key?: PublicKeyCredentialDescriptor) => {
+      if (!keyStore) {
+        throw new Error("Key store not initialized");
+      }
+      // Pass the keyName to PassKey.get() if provided
+      const passkey = await PassKey.get({ allowCredentials: key ? [key] : [] });
+      const root = await passkey.createRootKey();
+      await keyStore.set(ROOT_KEY, root);
+      setRoot(root);
+      return passkey;
+    },
+    [keyStore],
+  );
 
   const passphraseRegister = useCallback(async () => {
     // Don't store the root identity here. Return only the
     // mnemonic so that the UI can present guidance on handling
     // the private key. The root will be derived from the mnemonic
     // on authentication.
-    let [_, mnemonic] = await Identity.generateMnemonic();
+    const [_, mnemonic] = await Identity.generateMnemonic();
     return mnemonic;
   }, []);
 
-  const passphraseAuthenticate = useCallback(async (mnemonic: string) => {
-    if (!keyStore) {
-      return;
-    }
-    let root = await Identity.fromMnemonic(mnemonic);
-    await keyStore.set(ROOT_KEY, root);
-    setRoot(root);
-  }, [keyStore]);
-  
+  const passphraseAuthenticate = useCallback(
+    async (mnemonic: string) => {
+      if (!keyStore) {
+        return;
```

Comment:
Like `passkeyAuthenticate`, maybe this should throw if `!keyStore`, rather than fail silently -- if `!keyStore` then the initialization effect failed (no indexeddb maybe?)